erpnext/controllers/queries.py

```diff
@@ -414,10 +414,11 @@ def get_doctype_wise_filters(filters):
 
 @frappe.whitelist()
 def get_batch_numbers(doctype, txt, searchfield, start, page_len, filters):
```

Comment:
The PR in this state will introduce a regression.
### Before
![peek 2018-01-10 19-54](https://user-images.githubusercontent.com/818803/34790079-d9afa42a-f640-11e7-8793-98c9542fefbb.gif)

### After
![peek 2018-01-10 19-56](https://user-images.githubusercontent.com/818803/34790093-e2409f36-f640-11e7-8383-96dab0fc4743.gif)

### Stack trace
```(python
Traceback (most recent call last):
  File "/home/tunde/bench-repo/dev/apps/frappe/frappe/app.py", line 62, in application
    response = frappe.handler.handle()
  File "/home/tunde/bench-repo/dev/apps/frappe/frappe/handler.py", line 22, in handle
    data = execute_cmd(cmd)
  File "/home/tunde/bench-repo/dev/apps/frappe/frappe/handler.py", line 53, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "/home/tunde/bench-repo/dev/apps/frappe/frappe/__init__.py", line 939, in call
    return fn(*args, **newargs)
  File "/home/tunde/bench-repo/dev/apps/frappe/frappe/desk/search.py", line 14, in search_link
    search_widget(doctype, txt, query, searchfield=searchfield, page_length=page_length, filters=filters)
  File "/home/tunde/bench-repo/dev/apps/frappe/frappe/desk/search.py", line 35, in search_widget
    searchfield, start, page_length, filters, as_dict=as_dict)
  File "/home/tunde/bench-repo/dev/apps/frappe/frappe/__init__.py", line 939, in call
    return fn(*args, **newargs)
  File "/home/tunde/bench-repo/dev/apps/erpnext/erpnext/controllers/queries.py", line 424, in get_batch_numbers
    return frappe.db.sql(query)
  File "/home/tunde/bench-repo/dev/apps/frappe/frappe/database.py", line 176, in sql
    self._cursor.execute(query)
  File "/home/tunde/bench-repo/dev/env/local/lib/python2.7/site-packages/pymysql/cursors.py", line 165, in execute
    result = self._query(query)
  File "/home/tunde/bench-repo/dev/env/local/lib/python2.7/site-packages/pymysql/cursors.py", line 321, in _query
    conn.query(q)
  File "/home/tunde/bench-repo/dev/env/local/lib/python2.7/site-packages/pymysql/connections.py", line 860, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/home/tunde/bench-repo/dev/env/local/lib/python2.7/site-packages/pymysql/connections.py", line 1061, in _read_query_result
    result.read()
  File "/home/tunde/bench-repo/dev/env/local/lib/python2.7/site-packages/pymysql/connections.py", line 1349, in read
    first_packet = self.connection._read_packet()
  File "/home/tunde/bench-repo/dev/env/local/lib/python2.7/site-packages/pymysql/connections.py", line 1018, in _read_packet
    packet.check_error()
  File "/home/tunde/bench-repo/dev/env/local/lib/python2.7/site-packages/pymysql/connections.py", line 384, in check_error
    err.raise_mysql_exception(self._data)
  File "/home/tunde/bench-repo/dev/env/local/lib/python2.7/site-packages/pymysql/err.py", line 107, in raise_mysql_exception
    raise errorclass(errno, errval)
ProgrammingError: (1064, u"You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'Pork' at line 3")
```

Reply:
Fixed!